grails-datastore-gorm-neo4j/src/main/groovy/org/grails/datastore/gorm/neo4j/engine/DynamicAssociationSupport.java

```diff
@@ -38,6 +45,9 @@
         if(alreadyLoaded == null && hasDynamicAssociations) {
             session.setAttribute(object, Neo4jEntityPersister.DYNAMIC_ASSOCIATION_PARAM, Boolean.TRUE);
 
+            if (id == null) {
```

Comment:
I changed the code a bit in the Neo4jEntity to make this more clear. We reached this line, because dynamic associations were not loaded, then, in line 50, we flag the object as having dynamic the associations loaded and as the id is null we are done, and we just return.